includes/class-category-pager.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Newspack category pager fix.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Newspack category pager.
+ *
+ * This class adds tweaks to sites where the permalink structure is "/%category%/%postname%/".
+ *
+ * See https://core.trac.wordpress.org/ticket/8905 and https://core.trac.wordpress.org/ticket/21209
+ */
+final class Category_Pager {
+	/**
+	 * Initialize Hooks.
+	 */
+	public static function init() {
+		add_action( 'init', [ __CLASS__, 'on_init' ] );
+	}
+
+	/**
+	 * Action callback for init.
+	 */
+	public static function on_init() {
+		if ( '/%category%/%postname%/' === get_option( 'permalink_structure' ) ) {
```

Comment:
I've ran some tests and the issue also occurs with other settings, for example `/%category%/%post_id%/`. 

It does not happen if you do `/%post_id%//%category%` though.

Didn't go too far to get at the bottom of why this happens, but I think we can assume that it happens if the permalink structure starts with `category`?

Reply:
Good catch! I changed it to sniff if the permalink structure starts with `/%category%/`.